### ISSUE TITLE
KeyError: 0 when running `monitorcontrol --get-monitors`

### ISSUE THREAD

**Issue by JohnTravolski** [newAM/monitorcontrol#249]
I have a setup with 8 total monitors on Windows 10, Python 3.8, monitorcontrol==3.0.2.  When I run:

`monitorcontrol --get-monitors`

I get:

```
Monitor 1: PA278QV
Available Inputs:
        InputSource.DVI1
        InputSource.DP1
        InputSource.DP2
        InputSource.HDMI1*
Monitor 2: P2422H
Available Inputs:
        InputSource.ANALOG1
        InputSource.DP1
        InputSource.HDMI1*
Monitor 3: P2422H
Available Inputs:
        InputSource.ANALOG1
        InputSource.DP1
        InputSource.HDMI1*
Monitor 4: PA278QV
Available Inputs:
        InputSource.DVI1
        InputSource.DP1
        InputSource.DP2
        InputSource.HDMI1*
Monitor 5: PA278QV
Available Inputs:
        InputSource.DVI1
        InputSource.DP1
        InputSource.DP2
        InputSource.HDMI1*
Monitor 6: P2419H
Available Inputs:
        InputSource.ANALOG1*
        InputSource.DP1
        InputSource.HDMI1
Monitor 7: P2419H
Available Inputs:
        InputSource.ANALOG1*
        InputSource.DP1
        InputSource.HDMI1
Traceback (most recent call last):
  File "c:\program files\python\python38\lib\runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files\python\python38\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Python\Python38\Scripts\monitorcontrol.exe\__main__.py", line 7, in <module>
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\__main__.py", line 157, in main
    monitors_dict = monitor_obj.get_vcp_capabilities()
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\monitorcontrol.py", line 208, in get_vcp_capabilities
    res = _parse_capabilities(cap_str)
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\monitorcontrol.py", line 593, in _parse_capabilities
    caps_dict[key] = _convert_to_dict(_extract_a_cap(caps_str, key))
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\monitorcontrol.py", line 553, in _convert_to_dict
    result_dict[group].append(val)
KeyError: 0
```
It looks like it is failing when trying to retrieve information about monitor 8 (I assume, I'm just guessing). Monitor 8 is an HP monitor while all the other monitors are Dells and Asus monitors, in case that matters.

Any idea what to do here?

**Comment by newAM** [comment]
Can you show the output of `monitorcontrol --get-monitors -vvvv`?  My guess at the moment is a bug in the capabilities parser.

**Comment by JohnTravolski** [comment]
Sure:
```
C:\Users\username>monitorcontrol --get-monitors -vvvv
Monitor 1: PA278QV
Available Inputs:
        InputSource.DVI1
        InputSource.DP1
        InputSource.DP2
        InputSource.HDMI1*
Monitor 2: P2422H
Available Inputs:
        InputSource.ANALOG1
        InputSource.DP1
        InputSource.HDMI1*
Monitor 3: P2422H
Available Inputs:
        InputSource.ANALOG1
        InputSource.DP1
        InputSource.HDMI1*
Monitor 4: PA278QV
Available Inputs:
        InputSource.DVI1
        InputSource.DP1
        InputSource.DP2
        InputSource.HDMI1*
Monitor 5: PA278QV
Available Inputs:
        InputSource.DVI1
        InputSource.DP1
        InputSource.DP2
        InputSource.HDMI1*
Monitor 6: P2419H
Available Inputs:
        InputSource.ANALOG1
        InputSource.DP1
        InputSource.HDMI1*
Monitor 7: P2419H
Available Inputs:
        InputSource.ANALOG1
        InputSource.DP1
        InputSource.HDMI1*
Traceback (most recent call last):
  File "c:\program files\python\python38\lib\runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files\python\python38\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Python\Python38\Scripts\monitorcontrol.exe\__main__.py", line 7, in <module>
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\__main__.py", line 157, in main
    monitors_dict = monitor_obj.get_vcp_capabilities()
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\monitorcontrol.py", line 208, in get_vcp_capabilities
    res = _parse_capabilities(cap_str)
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\monitorcontrol.py", line 593, in _parse_capabilities
    caps_dict[key] = _convert_to_dict(_extract_a_cap(caps_str, key))
  File "c:\program files\python\python38\lib\site-packages\monitorcontrol\monitorcontrol.py", line 553, in _convert_to_dict
    result_dict[group].append(val)
KeyError: 0
```

**Comment by JohnTravolski** [comment]
I dug a bit more into it and started printing the `caps_str` variable in the _convert_to_dict function. This is what i get for the HP monitor that has the problem:
```
01 02 03 07 0C E3 F3
02 04 08 0B 0C 10 12 16 18 1A 2E(00 01 02 03) 52 60(01 0F 11) 6C 6E 70 86(02 05) 87(01 02 03 04 05 06 07) B2(00) C0 C6 C8 C9 CA(01 02) CC(01 02 03 04 05 06 0A 0D 0E 14) D6(01 02 03 04 05) DC(00(00 12 13 14 15 1D 1E)) DF E0(02(00 01 02)) E1(04(00 01 03 04 06 08 09) 0A(00 01) 0F(00 01) 10(00 01) 27(02 03 04 05 07 09 0A FF) 36(00 01) 3B 3C 3E 3F(02 03 04 05) 40(00 01 02 03) 42(00 01) 43(00 01) 45(00 01) 4A(00 01) 4C(00 01) 4E(00 01 02 03 04 05 FF) 56) E6(00 01) E7(00 01) E8(18(00 01) 1E(00 01 02)) EA(00 01) EB(00(01 02 03 04 05)) EE(01 02 03) EF(01 02 03 04 05) F0 F5(00 01 02 03 05 06 0A 0C 0D 0E 0F 11 12 13 14 15 1C 1E) FA(01 02) FB FC FD FE(00 01 02 04)
```
I notice it is a lot longer than it is for the Asus and Dell monitors.

**Comment by newAM** [comment]
Thanks, that is the string I was looking for :)

Turns out I forgot to add logging to the Windows implementation 🤦‍♂️

I'm pretty busy lately, I'll take a deeper look at this next weekend.

**Comment by JohnTravolski** [comment]
> 

Any chance you've been able to take a look at this?

**Comment by newAM** [comment]
Sorry about that!  My GitHub inbox has been crazy lately and I forgot about this issue under a pile of others, thanks for the reminder!

I have opened #256 which should fix this (and adds a test to make sure it stays working in the future), the problem area was this style of capabilities `DC(00(00 12 13 14 15 1D 1E))` I did not know groups could be nested like that when I fixed a parser bug for #110

Let me know if you want to test that, or I can merge it and make a release.
